src/hotspot/share/opto/type.cpp

```diff
@@ -710,6 +712,69 @@ void Type::Initialize_shared(Compile* current) {
   mreg2type[Op_VecY] = TypeVect::VECTY;
   mreg2type[Op_VecZ] = TypeVect::VECTZ;
 
+  LockNode::lock_type_init();
+  OptoRuntime::new_instance_Type_init();
```

Comment:
I suggest to move the initialization code into `OptoRuntime`. As a benefit, you'll be able to directly access fields from there, so some trivial `init` methods won't be needed anymore.

Reply:
"first" call is made from here because of shared space. Otherwise the object-allocation will deleted and VM will crash. That's what I observed. And again that was the reason why the initialization call is made from `Type::Initialize_shared`.

Reply:
My suggestion is about refactoring the code, so initialization is performed in `OptoRuntime` code (e.g., in `OptoRuntime::initialize_types()`). Then you call it from here.

Reply:
@iwanowww can you take another look. I have pushed the requested changes.